File: src_python/trading/exchange_router.py
from __future__ import annotations

from src_python.trading.deribit_exchange import DeribitExchange
from src_python.trading.derive_exchange import DeriveExchange
from src_python.trading.option_exchange import OptionsExchange

_exchange_map: dict[str, OptionsExchange] | None = None


def get_exchange(name: str) -> OptionsExchange:
    global _exchange_map
    if _exchange_map is None:
        _exchange_map = {
            "deribit": DeribitExchange(),
            "derive": DeriveExchange(),
        }

    exchange = _exchange_map.get(name.lower())
    print(name, exchange)
    if exchange is None:
        raise ValueError(f"Exchange not supported: {name}")
    return exchange

File: src/trading/ExchangeRouter.ts
import { DeribitExchange } from "./DeribitExchange";
import { DeriveExchange } from "./DeriveExchange";
import { OptionsExchange } from "./OptionExchange";

const exchangeMap: Record<string, OptionsExchange> = {
  deribit: new DeribitExchange(),
  derive: new DeriveExchange(), // temporary
};

export function getExchange(name: string): OptionsExchange {
  const ex = exchangeMap[name.toLowerCase()];
  console.log(name, ex);

  if (!ex) throw new Error(`Exchange not supported: ${name}`);
  return ex;
}
